aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -549,10 +545,25 @@ impl AggregatorData {
                 .get_mut(&AggregatorVersionedID::V2(base_aggregator))
             {
                 Some(aggregator) => {
-                    let value = aggregator.read_aggregated_aggregator_value(resolver)?;
-                    Ok(SnapshotValue::Integer(expect_ok(
-                        BoundedMath::new(aggregator.max_value).unsigned_add_delta(value, &delta),
-                    )?))
+                    // We need to make sure speculative_start_value is in a state for read,
+                    // but need to take it directly aftewrards, not the aggregated value returned by this function
+                    aggregator.read_aggregated_aggregator_value(resolver)?;
```

Comment:
Oh, so we have been applying delta twice? I think this is very ugly, can we make `read_aggregated_aggregator_value` return () and make sure it can be called on delta only?

Btw, isn't this a fix?
```
let value = aggregator.read_aggregated_aggregator_value(resolver)?;
 Ok(SnapshotValue::Integer(value))
```
Here you are basically copying what `read_aggregated_aggregator_value` is doing.

Reply:
Delta in the aggregator and in the snapshot can be different.
I.e. we could do +2, then snapshot, then +3.
snapshot will have +2, aggregator will have +5.

I intentionally don't have speculative_value inside of snapshot, as when fetching a new value, checking it is still valid for the full history (i.e. for the +5 on the aggregator) is the correct thing to do.

read_aggregated_aggregator_value is directly used for aggregator reads. I was thinking of having a boolean/enum additional argument to that function, to decide what to return - whether a starting or ending value. if that sounds better, I can update